DataFormats/Reconstruction/include/ReconstructionDataFormats/MatchInfoHMP.h

```diff
@@ -0,0 +1,69 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file MatchInfoHMP.h
+/// \brief Class to store the output of the matching to HMPID
+
+#ifndef ALICEO2_MATCHINFOHMP_H
+#define ALICEO2_MATCHINFOHMP_H
+
+#include "ReconstructionDataFormats/TrackLTIntegral.h"
+#include "ReconstructionDataFormats/GlobalTrackID.h"
+#include "CommonDataFormat/EvIndex.h"
+
+namespace o2
+{
+namespace dataformats
+{
+class MatchInfoHMP
+{
+  using GTrackID = o2::dataformats::GlobalTrackID;
+
+ public:
+  MatchInfoHMP(int idxHMPClus, GTrackID idxTrack, float angle = 0, float q = 0, float size = 0, std::vector<float>* photsClusCharge = nullptr) : mIdxHMPClus(idxHMPClus), mIdxTrack(idxTrack), mCkovAngle(angle), mMipCluQ(q), mMipCluSize(size), mPhotsClusCharge(photsClusCharge){};
+  MatchInfoHMP() = default;
+
+  void setIdxHMPClus(int index) { mIdxHMPClus = index; }
+  int getIdxHMPClus() const { return mIdxHMPClus; }
+
+  void setIdxTrack(GTrackID index) { mIdxTrack = index; }
+  GTrackID getTrackRef() const { return mIdxTrack; }
+
+  int getTrackIndex() const { return mIdxTrack.getIndex(); }
+
+  void setCkovAngle(float angle) { mCkovAngle = angle; }
+  float getCkovAngle() const { return mCkovAngle; }
+
+  void setMipClusQ(float q) { mMipCluQ = q; }
+  float getMipClusQ() const { return mMipCluQ; }
+
+  void setMipClusSize(int size) { mMipCluSize = size; }
+  int getMipClusSize() const { return mMipCluSize; }
+
+  void setNPhots(int n) { mNPhots = n; }
+  int getNPhots() const { return mNPhots; }
+
+  void print() const;
+
+ private:
+  int mIdxHMPClus;    // Idx for HMP cluster
+  GTrackID mIdxTrack; // Idx for track
+  float mCkovAngle;
+  float mMipCluQ = 0.0;                 // MIP cluster charge
+  int mMipCluSize = 0.0;                // MIP cluster size
+  int mNPhots = 0.0;                    // number of candidate photo-electrons
+  std::vector<float>* mPhotsClusCharge; // list of cluster charge of candidate photo-electrons
```

Comment:
we are trying to avoid dynamical size data formats, moreover with raw pointer, such objects are not messageable via shared memory and need require expensive root streaming. 
Do you really need a vector of charges? In such cases we usually add yet another vector which would keep the charges and the match would just keep the 1st entry index and the number of entries it has in this vector.

Reply:
It refers to the charge of the cherenkov photon clusters for the given track. This is a information very useful and I would like to not lose it! So adding an other vector, but it is not clear to me where?

Reply:
E.g. HMPClusterCharges via
```
auto getHMPClusterCharges() const { return getSpan<float>(GTrackID::HMP, PATTERNS); }
```
The idea is to refer to the range of entries in this vector from the HMPCluste in the same way as you refer from HMPTriggers to HMPClusters.